### ISSUE TITLE
scss-lint:disable command hides lints in other files

### ISSUE THREAD

**Issue by srawlins** [brigade/scss-lint#429]
I've found that using `scss-lint:disable` at the top of one file can affect what lints are filtered in other files. Also:

* If the file with `scss-lint:disable` is _n_ lines long, it _doesn't_ filter lints in other files on line numbers greater than _n_.
* The file with `scss-lint:disable` only affects other files when it is specified on the command line _after_ another file.

Examples:

**a.scss**

```scss
foo {
  border: 0px;

  // This next lint is on line 5; c.scss is 5 lines long.
  margin: 0px;
}
```

**c.scss**

```scss
// scss-lint:disable ZeroUnit

bar {
  border: 0;
}
```

```shell
$ bundle exec ../bin/scss-lint a.scss
a.scss:2 [W] ZeroUnit: `0px` should be written without units as `0`
a.scss:5 [W] ZeroUnit: `0px` should be written without units as `0`

$ bundle exec ../bin/scss-lint a.scss c.scss
a.scss:5 [W] ZeroUnit: `0px` should be written without units as `0`

$ bundle exec ../bin/scss-lint c.scss a.scss
a.scss:2 [W] ZeroUnit: `0px` should be written without units as `0`
a.scss:5 [W] ZeroUnit: `0px` should be written without units as `0`
```

**Comment by sds** [comment]
Thanks for the clear bug report with reproducible steps, @srawlins!

Fixed in 85f6630f33e0aa1e5e4ce01f632133eeec589a3f.
